### PR TITLE
Refactor import streams and add tests

### PR DESCRIPTION
**Pull request by orangejulius**
This is a big, and almost pure refactoring of all the tests and streams involved in the Openaddresses importer.

## Changes
* invalid CSV rows are filtered out in a separate stream
* code to perform cleanup on record fields is now called only from a separate cleanup stream
* the one giant test file is split into files for testing different areas of the code
* a little bit more explicit handling of whitespace trimming is added: previously values that consisted only of whitespace (possible since the CSV parser's trim functionality, which we are using, doesn't trim quoted CSV fields) would be caught by pelias-model. They are now explicitly filtered out before then
* doubly nested try/catch blocks are removed or replaced with better control flow
* a couple additional test cases are added to the functional tests
* Add logging for invalid CSV rows

Fixes pelias/pelias#228

### CODE REVIEW

File: lib/import_pipelines.js [hunk 1]
```diff
@@ -12,20 +12,75 @@ var through = require( 'through2' );
...
 }
 
-/**
- * Used to track the UID of individual records passing through the stream
- * created by `createRecordStream()`. Not bothering with a closure for
- * readability. See `peliasModel.Document.setId()` for information about UIDs.
+/*
+ * Create a through2 stream to filter out invalid records
+ */
+function createValidRecordFilterStream() {
```
Comment (dianashk) [inline lib/import_pipelines.js]: I think it generally reads cleaner if each stream is in its own module, even if really simple.

**Comment by dianashk** [comment]
Given that there are already 2 other PRs waiting on top of this, it probably doesn't really need to be refactored into one stream per module at this time, unless @orangejulius you think it wouldn't be a huge pain to merge downstream. Anyway, other than that comment it looks good to me.

**Comment by orangejulius** [comment]
That's a good point. The two PRs are actually built on top of each other (I just separated them out to keep them focused, two small PRs is better than one big one). I can just tack moving the streams to separate modules at the end.

**Comment by dianashk** [comment]
Perfect! Thanks.

**Comment by orangejulius** [comment]
Subsumed by #34

**Pull request closed without merging**